control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -282,6 +283,18 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
 
 	broker.Status.Address = addressableStatus.Address
 	broker.Status.Addresses = addressableStatus.Addresses
+
+	if feature.FromContext(ctx).IsOIDCAuthentication() && broker.Status.Address != nil {
+		audience := auth.GetAudience(eventing.SchemeGroupVersion.WithKind("Broker"), broker.ObjectMeta)
+		logging.FromContext(ctx).Debugw("Setting the brokers audience", zap.String("audience", audience))
+		broker.Status.Address.Audience = &audience
```

Comment:
can you also set the audience in `broker.Status.Addresses[*].Audience`? (and clear/nil them too if OIDC is disabled)